tests/js/function-caller.test.js

```diff
@@ -105,14 +94,20 @@ describe('Test functionCaller', () => {
 
 
   test('will not return AWS S3 Key if return value from callable not supplied', async () => {
+    // TODO: check S3 response object
+
+    let testKey = 'test/key';
 
-    AWS.S3.upload = jest.fn();
+    AWSMock.mock("S3", "upload", {Key: testKey});
     callable.mockReturnValue(undefined);
+    console.log = jest.fn();
 
-    expect.assertions(2);
     await functionCaller();
+
     expect(callable.mock.calls.length).toBe(1);
-    expect(AWS.S3.upload.mock.calls.length).toBe(0);
+
+    let logCallsLength = console.log.mock.calls.length - 1;
```

Comment:
is it the length or is the the last call? `lastLogCallIndex`?

Reply:
good call. Updated.